src/Network/Network/VirtualNetworkGateway/NewAzureVirtualNetworkGatewayCommand.cs

```diff
@@ -649,7 +649,7 @@ private PSVirtualNetworkGateway CreateVirtualNetworkGateway()
 
             if (!string.IsNullOrEmpty(this.GatewaySku) && this.GatewaySku.Equals(MNM.VirtualNetworkGatewaySkuTier.ErGwScale))
             {
-                if (this.MaxScaleUnit > 0 && this.MinScaleUnit > this.MaxScaleUnit)
+                if (this.MaxScaleUnit > 0 && this.MinScaleUnit > this.MaxScaleUnit || this.MinScaleUnit > 40 || this.MaxScaleUnit > 40)
```

Comment:
min > 40 || max > 40
can you please confirm whether this is correct?

Reply:
I've appropriately modified this

Reply:
I mean, you ensured min <= max already, why comparing min with 40? max <= 40 guarantees min <= 40

Reply:
OK I updated it again. Now I will just throw an error if max >= 40